src/CattyTests/DataModel/UserVariableTests.swift

```diff
@@ -70,4 +73,91 @@ final class UserVariableTests: XCTestCase {
+        XCTAssertTrue(userVariableA.isEqual(to: userVariableB))
+        XCTAssertEqual(userVariableB.value as! NSMutableArray, self.arrayWithDifferentElement)
+
+        let userVariableC = UserVariable()
+        userVariableC.name = "userVarC"
+
+        userVariableB = userVariableC
+        XCTAssertFalse(userVariableA.isEqual(to: userVariableB))
+    }
+
+    func testIsEqualToUserVariableForInitWithNameAndIsList() {
+        let userVariableA = UserVariable()
+        userVariableA.name = "userVar"
+        userVariableA.value = self.arrayWithSameElement
+
+        var userVariableB = UserVariable(name: userVariableA.name, isList: userVariableA.isList)
+        XCTAssertFalse(userVariableB!.isEqual(to: userVariableA))
+        XCTAssertNil(userVariableB?.value)
+
+        userVariableB?.value = userVariableA.value
+        XCTAssertTrue(userVariableB!.isEqual(to: userVariableA))
+        XCTAssertEqual(userVariableB?.value as! NSMutableArray, self.arrayWithSameElement)
+
+        userVariableA.value = self.arrayWithDifferentElement
+        userVariableB?.value = userVariableA.value
+        XCTAssertTrue(userVariableA.isEqual(to: userVariableB))
+        XCTAssertEqual(userVariableB?.value as! NSMutableArray, self.arrayWithDifferentElement)
+
+        userVariableB = UserVariable(name: userVariableA.name, isList: !userVariableA.isList)
+        XCTAssertFalse(userVariableB!.isEqual(to: userVariableA))
+    }
+
+    func testIsEqualToUserVariableForInitWithUserVariable() {
+        let userVariableA = UserVariable()
+        userVariableA.name = "userVar"
+        userVariableA.value = self.arrayWithSameElement
+
+        var userVariableB = UserVariable(variable: userVariableA)
+        XCTAssertFalse(userVariableB!.isEqual(to: userVariableA))
+        XCTAssertNil(userVariableB?.value)
+
+        userVariableB?.value = userVariableA.value
+        XCTAssertTrue(userVariableB!.isEqual(to: userVariableA))
+        XCTAssertEqual(userVariableB?.value as! NSMutableArray, self.arrayWithSameElement)
+
+        userVariableA.value = self.arrayWithDifferentElement
+        userVariableB?.value = userVariableA.value
+        XCTAssertTrue(userVariableA.isEqual(to: userVariableB))
+        XCTAssertEqual(userVariableB?.value as! NSMutableArray, self.arrayWithDifferentElement)
+
+        let userVariableC = UserVariable()
+        userVariableC.name = "userVarC"
+
+        userVariableB = UserVariable(variable: userVariableC)
+        XCTAssertFalse(userVariableA.isEqual(to: userVariableB))
+    }
+
+    func testIsEqualToUserVariableForMutableCopy() {
+        let userVariableA = UserVariable()
+        userVariableA.name = "userVar"
+        userVariableA.value = self.arrayWithSameElement
+
+        var userVariableB = userVariableA.mutableCopy(with: CBMutableCopyContext()) as! UserVariable
+        XCTAssertTrue(userVariableA.isEqual(to: userVariableB))
+        XCTAssertEqual(userVariableB.value as! NSMutableArray, self.arrayWithSameElement)
+
```

Comment:
Hi @m-herold  as you suggested to use the init to create a new userVariable. I am not sure if this test is required since it creates a mutable copy of uservariableA.